services/collectd/service.go

```diff
@@ -324,41 +347,38 @@ func (s *Service) writePoints() {
 	}
 }
 
-// Unmarshal translates a collectd packet into InfluxDB data points.
-func (s *Service) UnmarshalCollectd(packet *gollectd.Packet) []models.Point {
-	// Prefer high resolution timestamp.
-	var timestamp time.Time
-	if packet.TimeHR > 0 {
-		// TimeHR is "near" nanosecond measurement, but not exactly nanasecond time
-		// Since we store time in microseconds, we round here (mostly so tests will work easier)
-		sec := packet.TimeHR >> 30
-		// Shifting, masking, and dividing by 1 billion to get nanoseconds.
-		nsec := ((packet.TimeHR & 0x3FFFFFFF) << 30) / 1000 / 1000 / 1000
-		timestamp = time.Unix(int64(sec), int64(nsec)).UTC().Round(time.Microsecond)
-	} else {
-		// If we don't have high resolution time, fall back to basic unix time
-		timestamp = time.Unix(int64(packet.Time), 0).UTC()
-	}
+// Unmarshal translates a ValueList into InfluxDB data points.
+func (s *Service) UnmarshalValueList(vl *api.ValueList) []models.Point {
+	timestamp := vl.Time.UTC() // TODO: precision error?
```

Comment:
HR time is now implemented directly in the collectd/go-collectd library.
https://github.com/collectd/go-collectd/blob/master/network/parse.go#L168
https://github.com/collectd/go-collectd/blob/master/cdtime/cdtime.go#L26

The comment is bogus and relates to something else I fixed before submitting the original PR.